### ISSUE TITLE
Fix S2302 FP: Rule should be less sensitive for message of Argument exceptions

### ISSUE THREAD

**Issue by dmitry-golovinov-sonarsource** [SonarSource/sonar-dotnet#4454]
### Description

S2302 should be less sensitive to message inside ArgumentException, ArgumentNullException, ArgumentOutOfRangeException

### Repro steps

```
            throw new ArgumentException("This is argument."); // Noncompliant
            throw new ArgumentException("This is argument.", nameof(argument)); // Noncompliant FP
            throw new ArgumentException("argument and arg1"); // Noncompliant
            throw new ArgumentException("argument and arg1", nameof(arg1)); // Noncompliant FP
            throw new ArgumentNullException(nameof(argument), "Incorrect argument value"); // Noncompliant FP
            throw new ArgumentOutOfRangeException(nameof(argument), "Incorrect argument value"); // Noncompliant FP
            throw new ArgumentOutOfRangeException(nameof(argument), argument, "Incorrect argument value"); // Noncompliant FP
```

### Expected behavior

Issue should not be raised in the cases above.

### Actual behavior

Issue is being raised.

### Related information

* C#/VB.NET Plugins version 8.22
